OrbitVulkanLayer/TimerQueryPool.h

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_VULKAN_LAYER_TIMER_QUERY_POOL_H_
+#define ORBIT_VULKAN_LAYER_TIMER_QUERY_POOL_H_
+
+#include "OrbitBase/Logging.h"
+#include "absl/container/flat_hash_map.h"
+#include "absl/synchronization/mutex.h"
+#include "vulkan/vulkan.h"
+
+namespace orbit_vulkan_layer {
+
+namespace internal {
+enum SlotState {
+  kReadyForQueryIssue = 0,
+  kQueryPendingOnGPU,
+};
+}
+
+/*
+ * This class wraps Vulkan's VkQueryPool specific for timestamp queries, and provides utility
+ * methods to (1) initialize a poo, (2) retrieve an available slot index and (3) reset slot indices.
+ * In order to do so, it stores the internal `SlotState` for each index.
+ *
+ * Thread-Safety: This class is internally synchronized (using read/write locks) and can be safely
+ * accessed from different threads.
+ */
+template <class DispatchTable>
+class TimerQueryPool {
+ public:
+  explicit TimerQueryPool(DispatchTable* dispatch_table, uint32_t num_timer_query_slots)
+      : dispatch_table_(dispatch_table), num_timer_query_slots_(num_timer_query_slots) {}
+
+  /*
```

Comment:
Really nitpicking here: Do we have a convention for comments. I think we use 
```
// bla
// bla
// bla
```

instead of
```
/* blub
 * blub
 * blub */
```

Reply:
Done for all the method level comments.